packages/components/src/Popover/Popover.tsx

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable no-null/no-null */
+import React, { useState } from "react";
+import { usePopper } from "react-popper";
+import classnames from "classnames";
+import classes from "./Popover.css";
+import { Typography } from "../Typography";
+import { ButtonDismiss } from "../ButtonDismiss";
+
+export interface PopoverProps {
+  /**
+   * Element the popover will attach to and point at.
+   */
+  readonly attachTo: Element | React.RefObject<Element | null>;
+
+  /**
+   * Pop-over content.
+   */
+  readonly children: React.ReactNode;
+
+  /**
+   * Whether or not the popover is dismissable via the header close button.
+   */
+  readonly dismissible?: boolean;
+
+  /**
+   * Control popover viability.
+   */
+  readonly open?: boolean;
+
+  /**
+   * Callback executed when the user wants to close/dismiss the popover
+   */
+  readonly onRequestClose?: () => void;
+
+  /**
+   * Describes the preferred placement of the popper.  The first element in the placement array will be taken
+   * as the preferred location with the rest of the array as fallback.
+   */
+  readonly preferredPlacement?: "top" | "bottom" | "left" | "right" | "auto";
+
+  /**
+   * The title of the popover
+   */
+  readonly title: string;
+}
+
+export function Popover({
+  onRequestClose,
+  children,
+  dismissible = true,
+  attachTo,
+  open,
+  preferredPlacement = "auto",
+  title,
+}: PopoverProps) {
+  const [popperElement, setPopperElement] = useState<HTMLElement | null>(null);
+  const [arrowElement, setArrowElement] = useState<HTMLElement | null>(null);
+  const { styles: popperStyles, attributes } = usePopper(
+    attachTo instanceof Element ? attachTo : attachTo.current,
+    popperElement,
+    {
+      modifiers: buildModifiers(arrowElement),
```

Comment:
The `usePopper` internals does do some recomputation (https://github.com/popperjs/react-popper/blob/master/src/usePopper.js#L113) on each render if the list of modifiers has changed. (Which in our case will be so for each render). I am not really concerned about any performance issues but this is something to keep in mind if we do run into that in the future.